tools/acorn-optimizer.mjs

```diff
@@ -2029,16 +2033,22 @@ let extraInfo = null;
 if (extraInfoStart > 0) {
   extraInfo = JSON.parse(input.substr(extraInfoStart + 14));
 }
-// Collect all JS code comments to this array so that we can retain them in the outputted code
-// if --closureFriendly was requested.
-const sourceComments = [];
+// Collect all JS code comments to this map so that we can retain them in the
+// outputted code if --closureFriendly was requested.
+const sourceComments = {};
+const currentComments = [];
 let ast;
 try {
   ast = acorn.parse(input, {
     // Keep in sync with --language_in that we pass to closure in building.py
     ecmaVersion: 2021,
     preserveParens: closureFriendly,
-    onComment: closureFriendly ? sourceComments : undefined,
```

Comment:
Looks like before we avoided collecting info here, if we would not use it later. Is that not worth still doing, for efficiency?

Reply:
Oh yes I can revert that part.

Reply:
Same think in `onToken`, I think? We don't need to build up `sourceComments/currentComments` if not `closureFriendly`.

Reply:
Refactored.